spec-0007/index.md

```diff
@@ -129,12 +120,37 @@ This is implemented using:
    It changes the documentation and auto-completion to only advertise the new `rng` parameter.
 
 ```python
-_NoValue = object()  # singleton to indicate not explicitly passed
+import numpy as np
+import functools
+import numbers
+import warnings
 
+_NoValue = object()  # singleton to indicate not explicitly passed
 
-def check_random_state(seed=_NoValue, rng=_NoValue):
+# Should this specify the version number in which the behavior will change?
+def _check_random_state(seed=_NoValue, rng=_NoValue):
+    """ PRNG validator during deprecation period
+    
+    Accepts inputs specified via old `seed` (or `random_state`) arguments
+    or new `rng` argument and returns appropriate PRNG.
+    
+    Once the removal of `seed`/`random_state` argument is executed,
+    this function may be removed, and `numpy.random.default_rng` may be
+    used to validate the `rng` argument.
+    
+    Parameters
+    ----------
+    seed : <insert long list of objects here>
```

Comment:
Maybe this just becomes "any", and the next one becomes `Generator, int, optional` or something like that.

Reply:
I figured I'd let you fix this part because you have been working on the types.

Reply:
`check_random_state` is gone; resolving!